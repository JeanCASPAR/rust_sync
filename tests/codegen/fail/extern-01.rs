use rustre::sync;

fn g(x: f64) -> i64 {
    x as i64 * 2
}

sync! {
    #![pass(3)]

    node non(c : bool) = (a)
    where
        a : float = merge c {
            true => 3.0 when c,
            false => 2.0 whennot c,
        };

    #[export]
    node oui(c : bool) = (d)
    where
        b : float = 0.0 -> spawn non(c),
        d : int = f(b);
}

fn main() {
    assert_eq!(
        sync::oui([true, false, true, false, true].into_iter().map(|x| (x,)))
            .map(|(d,)| d)
            .collect::<Vec<_>>(),
        [0, 6, 4, 6, 4],
    )
}
