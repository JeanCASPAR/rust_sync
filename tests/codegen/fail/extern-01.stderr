error[E0432]: unresolved import `super::f`
  --> tests/codegen/fail/extern-01.rs:21:19
   |
21 |         d : int = f(b);
   |                   ^
   |                   |
   |                   no `f` in the root
   |                   help: a similar name exists in the module: `g`
